src/main/java/org/jabref/gui/StateManager.java

```diff
@@ -140,6 +140,7 @@ public void setSelectedEntries(List<BibEntry> newSelectedEntries) {
 
     public void setSelectedGroups(BibDatabaseContext database, List<GroupTreeNode> newSelectedGroups) {
         Objects.requireNonNull(newSelectedGroups);
+        selectedGroups.clear();
```

Comment:
This is incorrect. Do not clear the entire map, as this will result in losing the selected groups when switching between open libraries.
```suggestion
```